core/iwasm/aot/aot_loader.c

```diff
@@ -1298,9 +1298,18 @@ load_table_init_data_list(const uint8 **p_buf, const uint8 *buf_end,
 #endif
         data_list[i]->offset.init_expr_type = (uint8)init_expr_type;
         data_list[i]->offset.u.i64 = (int64)init_expr_value;
-        data_list[i]->func_index_count = func_index_count;
-        read_byte_array(buf, buf_end, data_list[i]->func_indexes,
-                        (uint32)size1);
+        data_list[i]->value_count = value_count;
+        for (j = 0; j < data_list[i]->value_count; j++) {
+            data_list[i]->init_values[j].init_expr_type =
+                INIT_EXPR_TYPE_FUNCREF_CONST;
+#if defined(BUILD_TARGET_X86_64) || defined(BUILD_TARGET_AMD_64)            \
+    || defined(BUILD_TARGET_AARCH64) || defined(BUILD_TARGET_RISCV64_LP64D) \
+    || defined(BUILD_TARGET_RISCV64_LP64)
+            read_uint64(buf, buf_end, data_list[i]->init_values[j].u.ref_index);
+#else
+            read_uint32(buf, buf_end, data_list[i]->init_values[j].u.ref_index);
+#endif
+        }
```

Comment:
AOTTableInitData is refactored to sync up with the const expr stack mechanism. For compatibility with existing AoT module, the AoT file is not changed in non-GC mode (still store func indexes array), we need to read the uintptr_t and construct the init value one by one